cmd/sync/sync.go

```diff
@@ -3,64 +3,232 @@
+					log.Error("failed to publish message, reason: ", err)
+				}
+				// Restart on new message
+				continue
+			}
+			if !res.Valid() {
+				log.Errorf("result.error: %v", res.Errors())
+				log.Error("Validation failed")
+				// Nack message so the server gets notified that something is wrong but don't requeue the message
+				if e := delivered.Nack(false, false); e != nil {
+					log.Errorln("failed to Nack message, reason: ", e)
+				}
+
+				// Send the message to an error queue so it can be analyzed.
+				if e := mq.SendJSONError(&delivered, err.Error(), conf.Broker); e != nil {
+					log.Error("failed to publish message, reason: ", res.Errors())
+				}
+				// Restart on new message
+				continue
+			}
+
+			// we unmarshal the message in the validation step so this is safe to do
+			_ = json.Unmarshal(delivered.Body, &message)
+
+			// Extract the sha256 from the message and use it for the database
+			var checksumSha256 string
+			for _, checksum := range message.DecryptedChecksums {
+				if checksum.Type == "sha256" {
+					checksumSha256 = checksum.Value
+				}
+			}
+
+			c := completed{
+				User:               message.User,
+				Filepath:           message.Filepath,
+				AccessionID:        message.AccessionID,
+				DecryptedChecksums: message.DecryptedChecksums,
+			}
+
+			completeMsg, _ := json.Marshal(&c)
+
+			res, err = validateJSON(conf.SchemasPath, completeMsg)
+			if err != nil {
+				log.Errorf("json error: %v", err)
+				// Nack message so the server gets notified that something is wrong but don't requeue the message
+				if e := delivered.Nack(false, false); e != nil {
+					log.Errorln("failed to Nack message, reason: ", e)
+				}
+				// Send the message to an error queue so it can be analyzed.
+				if e := mq.SendJSONError(&delivered, err.Error(), conf.Broker); e != nil {
+					log.Error("failed to publish message, reason: ", err)
+				}
+				// Restart on new message
+				continue
+			}
+			if !res.Valid() {
+				log.Errorf("result.error: %v", res.Errors())
+				log.Error("Validation failed")
+				// Nack message so the server gets notified that something is wrong but don't requeue the message
+				if e := delivered.Nack(false, false); e != nil {
+					log.Errorln("failed to Nack message, reason: ", e)
+				}
+				// Send the message to an error queue so it can be analyzed.
+				if e := mq.SendJSONError(&delivered, err.Error(), conf.Broker); e != nil {
+					log.Error("failed to publish message, reason: ", err)
+				}
+				// Restart on new message
+				continue
+			}
+
+			log.Debug("Sync initiated")
+			file, err := inbox.NewFileReader(message.Filepath)
+			if err != nil {
+				log.Errorf("Failed to open file: %s, reason: %v", message.Filepath, err)
+				continue
+			}
+
+			dest, err := backup.NewFileWriter(message.Filepath)
+			if err != nil {
+				log.Errorf("Failed to create file, reason: %v", err)
+				continue
+			}
+
+			_, err = io.Copy(dest, file)
```

Comment:
Probably overkill, but we could consider adding ability to fetch archived size from database and verify that with the number of bytes copied.